api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCycleService.java

```diff
@@ -67,13 +73,13 @@ public PaymentCycle createAndSavePaymentCycleForEligibleClaim(Claim claim, Local
     /**
      * Update the PaymentCycle with the eligibility status and voucher entitlement and save to the db.
      *
-     * @param paymentCycle      The payment cycle
-     * @param eligibilityStatus The eligibility status
-     * @param entitlement       The entitlement
+     * @param paymentCycle The payment cycle
+     * @param decision     The decision containing details of the claimants eligibility and entitlement during this payment cycle
      */
-    public void updateAndSavePaymentCycle(PaymentCycle paymentCycle, EligibilityStatus eligibilityStatus, PaymentCycleVoucherEntitlement entitlement) {
-        paymentCycle.setEligibilityStatus(eligibilityStatus);
-        paymentCycle.applyVoucherEntitlement(entitlement);
+    public void updateAndSavePaymentCycle(PaymentCycle paymentCycle, EligibilityAndEntitlementDecision decision) {
```

Comment:
Should we have this method at all? My understanding of hibernate is that if we modify the PaymentCycle within a transaction the changes will be persisted - so we can just modify the PaymentCycle itself. Even if we do need to call the save method (e.g. if the cycle has been disconnected) I don't think we should have a method to which we need to keep adding parameters every time there's a change.

Reply:
I quite like this method, it makes it explicit what we're doing and at what point rather than waiting for Hibernate to do at magically at some point in time.